### ISSUE TITLE
:farmer: `CmdLine.ModelDownloadUnversioned` and `CollectionTest/DownloadCollectionTest` failing to find directory in Windows

### ISSUE THREAD

**Issue by Crola1702** [gazebosim/gz-fuel-tools#370]
## Environment
* OS Version: Windows
* Source gz-fuel_tools 8, 9 and ci-main

## Description
* Expected behavior: Test should pass
* Actual behavior: `CmdLine.ModelDownloadUnversioned` and `CollectionTest/DownloadCollectionTest` is failing because it doesn't find a directory

## Steps to reproduce
1. Run a build in https://build.osrfoundation.org/job/ign_fuel-tools-ci-win/
2. See test fail

## Output
Reference build: https://build.osrfoundation.org/job/ign_fuel-tools-ci-win/63/

Test regressions: 
* [CmdLine.ModelDownloadUnversioned](https://build.osrfoundation.org/job/ign_fuel-tools-gz-8-win/73/testReport/junit/(root)/CmdLine/ModelDownloadUnversioned/)
* [CollectionTest/DownloadCollectionTest.AllItems/0](https://build.osrfoundation.org/job/ign_fuel-tools-gz-8-win/73/testReport/junit/(root)/CollectionTest_DownloadCollectionTest/AllItems_0/)
* [CollectionTest/DownloadCollectionTest.AllItems/1](https://build.osrfoundation.org/job/ign_fuel-tools-gz-8-win/73/testReport/junit/(root)/CollectionTest_DownloadCollectionTest/AllItems_1/)
* [CollectionTest/DownloadCollectionTest.AllItems/2](https://build.osrfoundation.org/job/ign_fuel-tools-gz-8-win/73/testReport/junit/(root)/CollectionTest_DownloadCollectionTest/AllItems_2/)
* [CollectionTest/DownloadCollectionTest.Models/0](https://build.osrfoundation.org/job/ign_fuel-tools-gz-8-win/73/testReport/junit/(root)/CollectionTest_DownloadCollectionTest/Models_0/)
* [CollectionTest/DownloadCollectionTest.Models/1](https://build.osrfoundation.org/job/ign_fuel-tools-gz-8-win/73/testReport/junit/(root)/CollectionTest_DownloadCollectionTest/Models_1/)
* [CollectionTest/DownloadCollectionTest.Models/2](https://build.osrfoundation.org/job/ign_fuel-tools-gz-8-win/73/testReport/junit/(root)/CollectionTest_DownloadCollectionTest/Models_2/)

<details>
<summary>
ModelDownloadUnversioned output:
</summary>

```
7: [ RUN      ] CmdLine.ModelDownloadUnversioned
7: C:\J\workspace\ign_fuel-tools-ci-win\ws\gz-fuel-tools\src\gz_src_TEST.cc(190): error: Value of: this->stdErrBuffer.str().empty()
7:   Actual: false
7: Expected: true
7: [33m[Wrn] [C:\J\workspace\ign_fuel-tools-ci-win\ws\gz-fuel-tools\src\LocalCache.cc:105] [m[33mServer directory does not exist [[m[33mC:\J\workspace\ign_fuel-tools-ci-win\ws\build\gz-fuel_tools9\test_cache\fuel.ignitionrobotics.org[m[33m]
7: [m
7: [  FAILED  ] CmdLine.ModelDownloadUnversioned (600 ms)
```

</details>

<details>
<summary>
CollectionTests Log output:
</summary>

```
7: [  FAILED  ] CollectionTest/DownloadCollectionTest.AllItems/0, where GetParam() = 1 (3686 ms)
7: [ RUN      ] CollectionTest/DownloadCollectionTest.AllItems/1
7: C:\J\workspace\ign_fuel-tools-gz-8-win\ws\gz-fuel-tools\src\gz_src_TEST.cc(391): error: Value of: this->stdErrBuffer.str().empty()
7:   Actual: false
7: Expected: true
7: [33m[Wrn] [C:\J\workspace\ign_fuel-tools-gz-8-win\ws\gz-fuel-tools\src\LocalCache.cc:105] [m[33mServer directory does not exist [[m[33mC:\J\workspace\ign_fuel-tools-gz-8-win\ws\build\gz-fuel_tools8\test_cache\fuel.ignitionrobotics.org[m[33m]
7: [m[33m[Wrn] [C:\J\workspace\ign_fuel-tools-gz-8-win\ws\gz-fuel-tools\src\LocalCache.cc:105] [m[33mServer directory does not exist [[m[33mC:\J\workspace\ign_fuel-tools-gz-8-win\ws\build\gz-fuel_tools8\test_cache\fuel.ignitionrobotics.org[m[33m]
7: [m
7: [  FAILED  ] CollectionTest/DownloadCollectionTest.AllItems/1, where GetParam() = 2 (2333 ms)
```

</details>

This is a consistent error happening in gz-fuel_tools-win latest versions

Ratio for the last 15 days:

job_name|last_fail|first_fail|build_count|failure_count|failure_percentage
-- | -- | -- | -- | -- | --
ign_fuel-tools-gz-8-win|2023-09-04|2023-08-23|4|4|100.0
ign_fuel-tools-gz-9-win|2023-08-30|2023-08-30|3|1|33.33
ign_fuel-tools-ci-win|2023-08-27|2023-08-23|2|2|100.0

**Comment by claraberendsen** [comment]
This seems to have passed also to `ign_fuel-tools-gz-4-win`  even though no new code has been back ported; which suggest it's related to fuel-server changes.  

In this job started failing on Sept 19th and with a 75.0% in 3 out of 4 builds. 
The only build that passed is [![Build Status](https://build.osrfoundation.org/buildStatus/icon?job=ign_fuel-tools-ign-4-win&build=83)](https://build.osrfoundation.org/job/ign_fuel-tools-ign-4-win/83/)

**Comment by mjcarroll** [comment]
Please verify, but I believe this was fixed by #376.  The change was actually in gz-common which is why it appeared strangely in the version combos.
